apps/accounts/tests/test_views.py

```diff
@@ -109,6 +111,92 @@ def test_two_users_exist(self):
             views.find_user({'uid': '9999', 'email': 'you@amo.ca'})
 
 
+class TestRenderErrorHTML(TestCase):
+
+    def make_request(self):
+        return APIRequestFactory().get(reverse('accounts.authenticate'))
+
+    def login_error_url(self, **params):
+        return urlparams(reverse('users.login'), **params)
+
+    def render_error(self, error, next_path=None):
+        return views.render_error(
+            self.make_request(), error, format='html', next_path=next_path)
+
+    def test_error_no_code(self):
+        response = self.render_error(
+            views.ERROR_NO_CODE, next_path='/over/here')
+        assert response.status_code == 302
+        assert response['location'] == self.login_error_url(
+            to='/over/here', error=views.ERROR_NO_CODE)
+
+    def test_error_no_profile(self):
+        response = self.render_error(views.ERROR_NO_PROFILE)
+        assert response.status_code == 302
+        assert response['location'] == self.login_error_url(
+            error=views.ERROR_NO_PROFILE)
+
+    def test_error_state_mismatch(self):
+        response = self.render_error(
+            views.ERROR_STATE_MISMATCH,
+            next_path='https://www.google.com/')
+        assert response.status_code == 302
+        assert response['location'] == self.login_error_url(
+            error=views.ERROR_STATE_MISMATCH)
+
+    def test_error_user_mismatch(self):
+        response = self.render_error(
+            views.ERROR_USER_MISMATCH, next_path='/wat')
+        assert response.status_code == 302
+        assert response['location'] == self.login_error_url(
+            to='/wat', error=views.ERROR_USER_MISMATCH)
+
+    def test_error_user_migrated(self):
+        response = self.render_error(views.ERROR_USER_MIGRATED)
+        assert response.status_code == 302
+        assert response['location'] == self.login_error_url(
+            error=views.ERROR_USER_MIGRATED)
+
+
+class TestRenderErrorJSON(TestCase):
+
+    def setUp(self):
+        patcher = mock.patch('accounts.views.Response')
+        self.Response = patcher.start()
+        self.addCleanup(patcher.stop)
+
+    def make_request(self):
+        return APIRequestFactory().post(reverse('accounts.login'))
+
+    def render_error(self, error):
+        views.render_error(self.make_request(), error, format='json')
+
+    def test_error_no_code(self):
```

Comment:
I also don't think you need five tests here. You just need to test a couple states:
* render_error() shows a response with the error code
* render_error() raises a KeyError if a status code has not been defined for the error code